app/resources/v1/upload/api.py

```diff
@@ -9,27 +10,47 @@
 class UploadAPI(Resource):
 
     @login_required
-    def post(self):
+    def post(self, dirname):
+        """
+        Save an uploaded file or reject it if it is not a PDF.
+        
+        :param dirname: name of directory within the uploads directory in which to save the uploaded file
+        """
         files = request.files
         file_ = files[next(files.keys())]
-        filename = secure_filename(file_.filename)
-        upload_path = os.path.join(
-            current_app.config["UPLOAD_DIRECTORY"],
-            filename
-        )
-        file_.save(upload_path)
-        response_data = {
-            "name": filename,
-            "size": os.path.getsize(upload_path)
-        }
-        return api_response.success(response_data)
+        if self._file_is_pdf(file_):
+            upload_path = os.path.join(
+                current_app.config["UPLOAD_DIRECTORY"],
+                dirname
+            )
+            # upload_path = /uploads/123
+            if not os.path.exists(upload_path):
+                os.mkdir(upload_path)
+            filename = secure_filename(file_.filename)
+            upload_path = os.path.join(upload_path, filename)
+            # upload_path = /uploads/123/file.pdf
+            file_.save(upload_path)
+            response_data = {
+                "name": filename,
+                "size": os.path.getsize(upload_path)
+            }
+            return api_response.success(response_data)
+        else:
+            return api_response.fail("This is not a PDF file. Please remove.")
```

Comment:
I agree that we should run our error messages through the business owner.

The only argument I have for not putting the error messages in a constant file is that this message is only in this one place. If you think that setting up a constants file would be more preferable, I would be okay with that too.